src/test/java/uk/gov/justice/probation/courtcaseservice/service/mapper/CourtCaseMapperTest.java

```diff
@@ -1,33 +1,220 @@
+        assertThat(newEntity.getDefendantId()).isEqualTo(defendant.getDefendantId());
+        assertThat(newEntity.getId()).isNull();
+        assertThat(newEntity.getOffences()).hasSize(1);
+        assertThat(newEntity.getDefendantName()).isEqualTo(defendant.getDefendantName());
+        assertThat(newEntity.getName()).isEqualTo(defendant.getName());
+        assertThat(newEntity.getDefendantId()).isEqualTo(defendant.getDefendantId());
+        assertThat(newEntity.getType()).isEqualTo(defendant.getType());
+        assertThat(newEntity.getAddress()).isEqualTo(defendant.getAddress());
+        assertThat(newEntity.getCrn()).isEqualTo(defendant.getCrn());
+        assertThat(newEntity.getPnc()).isEqualTo(defendant.getPnc());
+        assertThat(newEntity.getCro()).isEqualTo(defendant.getCro());
+        assertThat(newEntity.getDateOfBirth()).isEqualTo(defendant.getDateOfBirth());
+        assertThat(newEntity.getSex()).isEqualTo(defendant.getSex());
+        assertThat(newEntity.getNationality1()).isEqualTo(defendant.getNationality1());
+        assertThat(newEntity.getNationality2()).isEqualTo(defendant.getNationality2());
+        assertThat(newEntity.getPreviouslyKnownTerminationDate()).isEqualTo(defendant.getPreviouslyKnownTerminationDate());
+        assertThat(newEntity.getSuspendedSentenceOrder()).isEqualTo(defendant.getSuspendedSentenceOrder());
+        assertThat(newEntity.getBreach()).isEqualTo(defendant.getBreach());
+        assertThat(newEntity.getPreSentenceActivity()).isEqualTo(defendant.getPreSentenceActivity());
+        assertThat(newEntity.getAwaitingPsr()).isEqualTo(defendant.getAwaitingPsr());
+        assertThat(newEntity.isManualUpdate()).isEqualTo(defendant.isManualUpdate());
+    }
+
+    @Test
+    void givenUpdateForSameCrn_whenCreateNew_thenPopulateAllFieldsAndUpdateProbationStatus() {
+
+        var defendant = EntityHelper.aDefendantEntity();
+
+        var newEntity = CourtCaseMapper.createDefendant(defendant, CRN, "CURRENT");
+
+        assertThat(newEntity).isNotSameAs(defendant);
+        assertThat(newEntity.getProbationStatus()).isEqualTo("CURRENT");
+    }
+
+    @Test
+    void whenCreateDefendantOffence_thenReturnNew() {
+
+        var offence = EntityHelper.aDefendantOffence();
+
+        var newEntity = CourtCaseMapper.createDefendantOffence(offence);
+
+        assertThat(newEntity).isNotSameAs(offence);
+        assertThat(newEntity.getId()).isNull();
+        assertThat(newEntity.getTitle()).isEqualTo(OFFENCE_TITLE);
+        assertThat(newEntity.getSummary()).isEqualTo(OFFENCE_SUMMARY);
+        assertThat(newEntity.getSequence()).isEqualTo(1);
+    }
+
+    @Test
+    void whenCreateHearings_thenReturnList() {
+
+        var hearingEntity1 = EntityHelper.aHearingEntity();
+        var hearingEntity2 = EntityHelper.aHearingEntity(SESSION_START_TIME.plusDays(1));
+
+        var hearings = CourtCaseMapper.createHearings(List.of(hearingEntity1, hearingEntity2));
```

Comment:
Would `clone/copyHearings` read better here and in these other cases where the new cases being created are id-less copies?

Reply:
Similar to the previous comment, - they are all cloning really minus the id.